src/v2/layouts/HomeScreen/ExistingRegistrationTable.jsx

```diff
@@ -0,0 +1,258 @@
+import DataGrid from '@pepcus/react-simple-data-grid';
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import isEqual from 'lodash/isEqual';
+import { updateMembersOptInApi } from './updateMembersOptInApi';
+import { existingRegistrationMetaData } from './existingRegistrationMetaData';
+import { AdminSecretKey } from '../../components/AdminSecretkey/secretKey';
+import { PrimaryButton } from '../../components/design/Button';
+import { CustomLoader } from '../../components/design/loader';
+import { Message } from '../../components/design/message';
+import { P1 } from '../../components/design/text/TextComponent';
+import { Form, Formik } from 'formik';
+import { CustomInput } from '../../components/design/form/CustomInputs';
+
+const ExistingRegistrationTable = ({ members, strings, onClickBackBtn }) => {
+	const {
+		MEMBER_DETAILS,
+		BACK,
+		SUBMIT,
+		OPT_IN_OR_OPT_OUT_SUCCESS_MESSAGE,
+		OPT_IN_OR_OPT_OUT_FAILED_MESSAGE,
+		NO_ALTERNATE_HINDI,
+		YES_HINDI,
+	} = strings;
+	const [isLoading, setIsLoading] = useState(false);
+	const [gridData, setGridData] = useState(members);
+	const [registrationSuccess, setRegistrationSuccess] = useState(null);
+	const [registrationError, setRegistrationError] = useState(null);
+	const [isSubmitDisable, setSubmitDisable] = useState(true);
+	const [initialGridData, setInitialGridData] = useState(members || []);
+	const { secretKey } = AdminSecretKey;
+
+	const replaceCustomComponentInMeta = () => {
+		const { headerConfig } = existingRegistrationMetaData;
+		return {
+			...headerConfig,
+			headerConfig: headerConfig?.map((hc) => {
+				if (hc?.key === 'optIn') {
+					return {
+						...hc,
+						customComponent: SwitchButton,
+					};
+				}
+				return hc;
+			}),
+		};
+	};
+
+	const SwitchButton = ({ rowData }) => {
+		const { id } = rowData;
+
+		const handleOnChange = (event) => {
+			setSubmitDisable(false);
+			setRegistrationSuccess(null);
+			setRegistrationError(null);
+			const updateData = (list) => {
```

Comment:
Looks like there is redundant function call
```suggestion
			const updatedData = gridData?.map((data) => {
```